### PR TITLE
using non blocking call if possible to generate randomness

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: daemon/src/common/System.cpp [hunk 1]
```diff
@@ -346,13 +354,23 @@ void GenRandomBytes(void* dest, size_t size)
 	if (nacl_secure_random(dest, size, &bytes_written) != 0 || bytes_written != size)
 		Sys::Error("nacl_secure_random failed");
 #else
+#ifdef __linux__
```
Comment (mbasaglia) [inline daemon/src/common/System.cpp]: Maybe it's more readable to have
```
///...
#elif defined(__linux__) && defined(HAS_GETRANDOM_SYSCALL)
//...
#elif defined(__linux__) 
//...
#endif
```

**Comment by DolceTriade** [comment]
Seems like the error here is somewhere else? This looks ok to me.

**Comment by Viech** [comment]
@DolceTriade what error are you talking about? Also, OK to merge now (also @mbasaglia)?

**Comment by mbasaglia** [comment]
TBH I'd just use <random> instead of all this custom code, but the changes themselves look ok to me.

**Comment by illwieckz** [comment]
Hi @devnexen! The _Dæmon Engine_ is now extracted from this source tree, you can find its dedicated repository there:

- https://github.com/DaemonEngine/Daemon

Can you replay your pull request on this repository? Merci et bienvenue ! :wink:

**Comment by devnexen** [comment]
Sure :-)

**Pull request closed without merging**